### ISSUE TITLE
Regarding the segmentation results

### ISSUE THREAD

**Issue by lianxxx** [facebookresearch/dino#54]
It's a so impressive work and thanks for your code. Regarding the results in the table of Figure 4, I want to ask how to generate the multi-label segmentation maps, ie, how to associate the self-attention maps to different classes.

**Comment by mathildecaron31** [comment]
Hi @lianxxx 

Thanks a lot for your interest and your kind words.

For a given image, I compare the mask obtain with each head against each ground-truth mask. For each GT mask, I pick the mask head with the best Jaccard distance. Does that make sense ?

My private code looks like this:
```
jac = 0
for o in objects:
    masko = mask == o
    intersection = masko * th_attn
    intersection = torch.sum(torch.sum(intersection, dim=-1), dim=-1)
    union = (masko + th_attn) > 0
    union = torch.sum(torch.sum(union, dim=-1), dim=-1)
    jaco = intersection / union
    jac += max(jaco)
jac /= len(objects)
jacs += jac

```

**Comment by lianxxx** [comment]
Yes, that makes sense. Thanks very much for your quick reply.

**Comment by vadimkantorov** [comment]
I processed some COCO val images through `visualize_attention.py`: (150Mb, OneDrive) https://1drv.ms/u/s!Apx8USiTtrYmqoY_ox2tgdtG7yOXeQ?e=g7yulY

@mathildecaron31 Do these look right to you?

**Comment by happyhackingit** [comment]
@vadimkantorov I got the same results on COCO val images. @mathildecaron31 wondering if you could share some insights.
